C/services/common/plugin_manager.cpp

```diff
@@ -225,3 +226,39 @@ PLUGIN_HANDLE PluginManager::resolveSymbol(PLUGIN_HANDLE handle, const string& s
   return pluginHandleMap.find(handle)->second->ResolveSymbol(symbol.c_str());
 }
 
+/**
+ * Scan "plugins" directory under FOGLAMP_ROOT for the given plugin type:
+ * south, north, filter, notificationRule, notificationDelivery etc
+ *
+ * @param    type		The plugin type
+ * @param    plugins		The output plugin list name to fill	
+ */
+void PluginManager::scanPluginDir(const string& type,
+				  list<string>& plugins)
+{
+	char *home = getenv("FOGLAMP_ROOT");
+	if (home)
+	{
+		struct dirent *entry;
+		DIR *dp;
+		string path = home;
+		path += "/plugins/" + type + "/";
+
+		// Open the plugins dir/type
+		dp = opendir(path.c_str());
+
+		// Get all plugin name directories
+		while ((entry = readdir(dp)))
+		{
+			if (strcmp (entry->d_name, "..") != 0 &&
+			    strcmp (entry->d_name, ".") != 0)
+			{
+				// Load plugin, given its name
+				loadPlugin(entry->d_name, type);
```

Comment:
This could fail if the file is not a valid plugin or is of the wrong type.

Load plugin does not take a filename as an argument, it takes a plugin name, they are not the same.